src/makeInfinityAdder.js

```diff
@@ -6,7 +6,27 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (value) => {
+    if (value === undefined && sum === 0) {
+      return 0;
+    }
+
+    if (value === undefined && sum !== 0) {
```

Comment:
You can combine these conditions. This way you will return `0` in case of sum being equal to `0`, and the `result` if the sum is not `0`
```suggestion
    if (value === undefined) {
```

Reply:
agree)) thanks! @glugovsky